### ISSUE TITLE
Misleading "There was a mismatch between the processor architecture of the project" warning (MSB3271)

### ISSUE THREAD

**Issue by Destructor-Ben** [bluemagic123/tModLoader#3361]
### Version

Latest 1.4-stable

### OS

Windows

### Platform

Steam

### This bug affects

Mod capability as a Modder

### Description

MSB3271 warning appears in the warnings section of Visual Studio warning about Steamworks.NET.dll
![Screenshot 2023-03-30 090446](https://user-images.githubusercontent.com/62684035/228654788-b6954626-b213-4038-bc0f-bc60e6f1fa08.png)

### Log File

Should be irrelevant but I can put it here if requested.

### Steps to reproduce

Open a mod project in Visual Studio
The warning should popup in the errors window

### Expected Behavior

No warning should appear

### Actual Behavior

MSB3271 warning appears in the warnings section of Visual Studio

### Reproduction frequency

Always

### Additional Information

This error should be completely harmless and the only harm it seems to do is clogging up the error window. It can be fixed by adding `<ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>None</ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch>`
to the .csproj. It is a minor error so I don't expect this to get fixed unless it is very simple. Is also an issue on 1.4 stable, 1.4 preview, and 1.4.4.

**Comment by Solxanich** [comment]
I can confirm this exact error appears with intel cpus as well. Not AMD cpu related.

Steamworks.net is kinda horrible library that is not proper AnyCPU and this is just another symptom of it. 
Not aware of any actual issues related to win x64 coming out of it.
Feel free to PR adding the warning suppressor since we probably aren't fixing the library any time soon

**Comment by Destructor-Ben** [comment]
Something Chicken-Bones mentioned was adding the warning suppressor to mods with libraries could have an issue like this, except it might actually be a problem, so adding this could prevent them from knowing.

**Comment by Chicken-Bones** [comment]
Yeah, we are intending to build a proper AnyCPU version of Steamworks.NET, but haven't done so yet.
